modules/network-annotation-processor/src/main/java/org/apache/ignite/network/processor/internal/AutoSerializableProcessor.java

```diff
@@ -0,0 +1,248 @@
+import org.apache.ignite.network.serialization.MessageSerializer;
+
+/**
+ * Annotation processor for generating (de-)serializers for network messages marked with the {@link AutoSerializable}
+ * annotation.
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_11)
+public class AutoSerializableProcessor extends AbstractProcessor {
+    /** {@inheritDoc} */
+    @Override public Set<String> getSupportedAnnotationTypes() {
+        return Set.of(AutoSerializable.class.getName());
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        Set<TypeElement> annotatedElements = annotations.stream()
+            .map(roundEnv::getElementsAnnotatedWith)
+            .flatMap(Collection::stream)
+            .map(TypeElement.class::cast)
+            .collect(Collectors.toUnmodifiableSet());
+
+        if (annotatedElements.isEmpty()) {
+            return true;
+        }
+
+        try {
+            generateSources(annotatedElements);
+        } catch (IOException e) {
+            throw new IllegalStateException("IO exception during annotation processing", e);
+        }
+
+        return true;
+    }
+
+    /**
+     * Generates serialization-related classes for the given elements.
+     */
+    private void generateSources(Set<TypeElement> annotatedElements) throws IOException {
+        var factories = new HashMap<TypeElement, TypeSpec>();
+
+        for (var messageClass : annotatedElements) {
+            try {
+                if (isValidElement(messageClass)) {
+                    String packageName = ClassName.get(messageClass).packageName();
+
+                    TypeSpec serializer = generateSerializer(messageClass);
+                    writeToFile(packageName, serializer);
+
+                    TypeSpec deserializer = generateDeseralizer(messageClass);
+                    writeToFile(packageName, deserializer);
+
+                    TypeSpec factory = generateFactory(messageClass, serializer, deserializer);
+                    writeToFile(packageName, factory);
+
+                    factories.put(messageClass, factory);
+                } else {
+                    processingEnv.getMessager().printMessage(
+                        Diagnostic.Kind.ERROR,
+                        String.format(
+                            "%s annotation must only be present on interfaces that extend %s",
+                            AutoSerializable.class, NetworkMessage.class
+                        ),
+                        messageClass
+                    );
+                }
+            } catch (ProcessingException e) {
+                processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, e.getMessage(), messageClass);
+            }
+        }
+
+        TypeSpec registryInitializer = generateRegistryInitializer(factories);
+        writeToFile(getParentPackage(annotatedElements), registryInitializer);
```

Comment:
What do you mean under "builder factory"?

Reply:
Factory that creates builder instances for messages

Reply:
we can have multiple factories in the same module, so I don't think it is possible in the general case

Reply:
I thought that factory signifies messages group, meaning that they have to be initialized together. So every factory may have its own initializer

Reply:
The idea was to have a single initializer per module, since this is the most convenient way (e.g. why would you want to initialize several groups separately?). Anyway, there's no such concept as a message group right now, this will likely change in the next ticket, when we will actually introduce message groups based on the `@MessageTypes` annotation, in which case there will indeed be an initializer per group (see the TODO comment in this method).